config/user-python.m4

```diff
@@ -0,0 +1,21 @@
+AC_DEFUN([ZFS_AC_CONFIG_USER_PYTHON_VERSIONS], [
+	AC_MSG_CHECKING([version of standard python application])
+
+	python_version=`/usr/bin/python -c "import sys; print (sys.version.split()[[0]])"`
```

Comment:
You won't be able to depend on `/usr/bin/python` in Fedora 29 and next versions. In Fedora 29 `/usr/bin/python` file will be packaged as a separate rpm not required in minimal installations. So you will end up with only `/usr/bin/python3`.

Reply:
That's going to be a problem for <a href="https://github.com/zfsonlinux/zfs/blob/master/cmd/arcstat/arcstat.py#L1">arcstat.py</a> as well then...

Reply:
Yes but a lot of "standard" packages using Python 2 will also depend on `/usr/bin/python` so in most cases if there will be `/usr/bin/python2` there also will be `/usr/bin/python`. At least in a few future releases. I'm just signaling a case of minimal/server installation where every utility is already ported to Python 3. This will spread to future RHEL and derivatives releases.

Reply:
@mskarbek thanks for the heads up.

@DeHackEd the autoconf infrastructure already exists to request a specific python version when compiling.  We should be able to leverage it to let the packaging request a specific version or when unspecified auto-detect the installed system version.  

```
$ ./configure --help
...
Some influential environment variables:
...
  PYTHON      the Python interpreter
  PYTHON_VERSION
              The installed Python version to use, for example '2.3'. This
              string will be appended to the Python interpreter canonical
              name.

Use these variables to override the choices made by `configure' or to help
it to find libraries and programs with nonstandard names/locations.
```

Reply:
I'd considered that, especially since the pyzfs code already makes use of the infrastructure there. But the `arc_summary.py`scripts already hardcodes that path in the shebang, so as is it is necessary for the name to exist as specified.